runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/hotcold/HotColdTemperature.java

```diff
@@ -111,4 +111,22 @@ public static HotColdTemperature getFromTemperatureSet(final Set<HotColdTemperat
 			.max(Comparator.comparingInt(x -> (x.getText()).length()))
 			.orElse(null);
 	}
+
+	/**
+	 * This is only used for dev mode through the ::hotcold command.
+	 * Builds the message we would get if we were a given distance from our target location
+	 */
+	public static String getMessageFromDistance(int distance, Set<HotColdTemperature> temps)
+	{
+		String message = DEVICE_USED_START_TEXT;
+		for (HotColdTemperature temp : temps)
+		{
+			if (distance >= temp.minDistance && distance <= temp.maxDistance)
```

Comment:
This looks like it would always just match the ice cold one?

Reply:
It doesn't. The distance passed in is the distance between the player and the spot we are looking for. If the distance is less than min it will check the max, and with both of those checks it can never match more than 1.

Example. Say that is 150. That is less than minDistance(500), so it moves and checks very cold. Also less than minDistance(200) so it checks cold. It is equal to minDistance(150) so it checks against max, which it is less than. So it returns cold.